synapse/types.py

```diff
@@ -235,8 +243,9 @@ class GroupID(DomainSpecificString):
     SIGIL = "+"
 
     @classmethod
-    def from_string(cls, s):
-        group_id = super(GroupID, cls).from_string(s)
+    def from_string(cls: Type[DS], s: str) -> DS:
+        group_id = super().from_string(s)  # type: DS # type: ignore
```

Comment:
Are we setting the type or ignoring it?

Reply:
Ha, this means "I promise mypy the result of this type is `str` now shut up about this line". `mypy` doesn't seem to like using `super` with a `Type[DS]`, which is why we have to ignore the line and manually type it.